adding regression test for modules

kingxleyy opened the pull request.

kingxleyy committed:
adding buildtest class

shahzebsiddiqui commented on tests/cli/test_build.py:
```diff
@@ -196,6 +196,41 @@ def test_build_buildspecs(self):
+                    timeout=None,
+                    limit=None,
+                    save_profile=None,
+                    profile=None,
```
why are you defining a `BuildTest` class this doesn't make sense. Just call the class and create an object

```python
cmd = Buildtest(...)
cmd.build()
```

kingxleyy committed:
adding buildtest class

kingxleyy committed:
Merge branch 'buildtesters:devel' into kingsley_buildtest

shahzebsiddiqui commented on tests/cli/test_build.py:
```diff
@@ -196,6 +196,7 @@ def test_build_buildspecs(self):
             exclude_buildspecs=[excluded_buildspecs[0]],
             buildtest_system=self.system,
         )
+        cmd = Buildtest(...)
```
i dont mean you literally copy the suggestion. The `...` was meant for you to fill the appropriate arguments to the class so you can invoke the BuildTest class. Please see examples from above. Once you are ready for review please ping me. In the meantime, i will mark this as a `Draft`

kingxleyy committed:
invoking buildtest

kingxleyy committed:
invoking buildtest

kingxleyy committed:
adding buildtest module

Mendi03 commented on tests/cli/test_build.py:
```diff
@@ -209,6 +209,14 @@ def test_build_buildspecs(self):
+    def test_module(self):
+        cmd = BuildTest(
+                configuration=configuration,
+                buildspecs=[buildspec_paths],
```
The regression test is failing because the variable `buildspec_paths` is empty.

Mendi03 reviewed (changes requested):
Check my comment on why the regression is failing. Furthermore, you still need to invoke the class **_with_** a value for `module` and `unload_modules`.

kingxleyy committed:
specifying buildtest modules and unload modules

kingxleyy committed:
specifying buildtest modules and unload modules

kingxleyy committed:
changing the comment to match the correct yml file name that will be linked for test

kingxleyy committed:
adding variables for buildspec_paths

Mendi03 commented on tests/cli/test_build.py:
see my previous comment

kingxleyy committed:
add values to arguments module and unload_module

shahzebsiddiqui committed:
change name of docstring and name of method
change invocation of "BuildTest" when invoking test and add comment on what is being tested
